pkg/v1/tkg/tkgconfigupdater/provider.go

```diff
@@ -75,18 +75,43 @@ func ensureCertManagerInConfig(defaultProviders providers, tkgConfigNode *yaml.N
 		certManagerIndex = GetNodeIndex(tkgConfigNode.Content[0].Content, constants.CertManagerConfigKey)
 	}
 
-	certManagerBytes, err := yaml.Marshal(defaultProviders.CertManager)
-	if err != nil {
+	certManagerNode := yaml.Node{}
+	if err := copyData(defaultProviders.CertManager, &certManagerNode); err != nil {
 		return errors.Wrap(err, "unable to get cert-manager")
 	}
 
-	certManagerNode := yaml.Node{}
-	err = yaml.Unmarshal(certManagerBytes, &certManagerNode)
+	tkgConfigNode.Content[0].Content[certManagerIndex] = certManagerNode.Content[0]
+	return nil
+}
+
+func copyData(from, to interface{}) error {
+	bytes, err := yaml.Marshal(from)
 	if err != nil {
-		return errors.Wrap(err, "unable to get cert-manager")
+		return err
+	}
+
+	switch v := to.(type) {
+	case *yaml.Node, *providers:
+		err = yaml.Unmarshal(bytes, v)
+		if err != nil {
+			return err
+		}
+	default:
+		return errors.New("unable to configure providers")
```

Comment:
There's no need for the `switch`. This would work just as well:

```go
 if err := yaml.Unmarshal(bytes, to); err != nil {
 	return err
 }
```

We can wrap the error where we're calling `copyData()`

Reply:
Done!